lib/Stan_math_backend.ml

```diff
@@ -163,19 +157,16 @@ let emit_error_wrapper ppf (emit_err, err_arg, emit_contents, contents_arg) =
   emit_str ppf " catch const std::exception& e) " ;
   emit_block ppf (emit_err, err_arg)
 
-let emit_runtime_exception ppf =
-  fprintf ppf "std::runtime_error(std::string(%s) + e.what())"
+let emit_located_msg ppf msg =
+  fprintf ppf
+    {|stan::lang::rethrow_located(
+    std::runtime_error(std::string(%s) + e.what(), current_statement__);
+// Next line prevents compiler griping about no return
+throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***");
+|}
+  @@ Option.value ~default:"e" msg
```

Comment:
What's going on with the formatting here?

Reply:
Not sure. Do you want me to move line 167 up to after the end of the multi-line string on line 166? That might pass the formatter, I forget. Just in case - the @@ is essentially `$` in Haskell - groups everything to its right as if we'd put parens around it.